examples/basic_authentication/routes/users/[id].dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:io';
+
+import 'package:dart_frog/dart_frog.dart';
+import 'package:example/user_repository.dart';
+
+Future<Response> onRequest(RequestContext context, String id) async {
+  return switch (context.request.method) {
+    HttpMethod.get => _getUser(context, id),
+    HttpMethod.patch => _updateUser(context, id),
+    HttpMethod.delete => _deleteUser(context, id),
+    _ => Future.value(Response(statusCode: HttpStatus.methodNotAllowed)),
+  };
+}
+
+Future<Response> _getUser(RequestContext context, String id) async {
+  final user = context.read<User>();
+  if (user.id != id) {
+    return Response(statusCode: HttpStatus.forbidden);
+  }
```

Comment:
You, as a user, can only get yourself? Should we add some authorization here? Maybe admin users can retrieve all other users?

Reply:
> Should we add some authorization here?

This is authorization code, in this example, only you can get your information, this was a design choice for the example to showcase the difference of authentication vs authorization, which I explain on the README.

> Maybe admin users can retrieve all other users?

Definitely, but I don't want to make this example unnecessarily complicated, this is an example for authentication, not authorization, and even so authorization was already showcased on the GET/PUT/POST /users as a bonus. An admin functionality could be a cool thing to be mentioned in a more complete tutorial thought.